### PR TITLE
Add Sensu Enterprise troubleshooting guide

### LINKED ISSUE #767: Possible causes of "undefined method get_private_key_info" error should be documented
## This is an issue with:

- [ ] Bug (site functionality or styling)
- [ ] Errata (fix needed for doc content)
- [ ] New content (guide wanted)
- [x] Update (Add missing or refresh existing content)

## Expected Behavior

Common errors should be documented

## Current Behavior

Sensu Enterprise users may commonly encounter the error "undefined method `get_private_key_info'" which typically arises due to invalid or encrypted SSL/TLS keys.

## Possible Solution

Add a troubleshooting guide to Sensu Enterprise documentation, document solutions for this error in that guide.

Example command for removing password/encryption from private keys:

`openssl rsa -in [encrypted.key] -out [plaintext.key]`

## Context

To help users quickly resolve this error, Sensu Enterprise 3.2 will add a more friendly message along with this error. We'd like the friendly message to provide a link to relevant documentation. See https://github.com/sensu/sensu-enterprise/issues/385

### PR DESCRIPTION
**Pull request by apaskulin**
## Description

Adds a troubleshooting guide for Sensu Enterprise that includes some basic log and configuration steps as well as the private key error that will link users to this page as of version 3.2.

To do:
- [ ] Copy to all versions post-review

## Motivation and Context

Closes #767

### CODE REVIEW

File: content/sensu-enterprise/3.2/guides/troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+---
+
+This guide provides an introduction to troubleshooting Sensu Enterprise.
+For information about troubleshooting the Sensu client and RabbitMQ connection, see the [Sensu Core troubleshooting guide][1].
+
+- [Reloading Configuration](#reloading-configuration)
+- [Collecting Logs](#collecting-logs)
+- [Settings Log Levels](#setting-log-levels)
+- [Printing Configuration](#printing-configuration)
+- [Resolving Private Key Errors](#resolving-private-key-erors)
```
Comment (rgeniesse) [inline content/sensu-enterprise/3.2/guides/troubleshooting.md]: should be `errors` on the end.

File: content/sensu-enterprise/3.2/guides/troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+   parent: guides
+---
+
+This guide provides an introduction to troubleshooting Sensu Enterprise.
+For information about troubleshooting the Sensu client and RabbitMQ connection, see the [Sensu Core troubleshooting guide][1].
+
+- [Reloading Configuration](#reloading-configuration)
+- [Collecting Logs](#collecting-logs)
+- [Settings Log Levels](#setting-log-levels)
+- [Printing Configuration](#printing-configuration)
```
Comment (rgeniesse) [inline content/sensu-enterprise/3.2/guides/troubleshooting.md]: Heading down below is `Printing Configurations`.

File: content/sensu-enterprise/3.2/guides/troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+{{< highlight shell >}}
+/var/log/sensu
+├── sensu-enterprise.log
+└── sensu-enterprise-dashboard.log{{< /highlight >}}
+
+You can view the Sensu Enterprise logs at the paths above, or provide them in an archive:
+
+{{< highlight shell >}}
+tail -n 10000 /var/log/sensu/sensu-enterprise.log > sensu-enterprise-10k.log && gzip -9 sensu-enterprise-10k.log{{< /highlight >}}
```
Comment (rgeniesse) [inline content/sensu-enterprise/3.2/guides/troubleshooting.md:49]: maybe call out above that this is only the last 10000 lines?

File: content/sensu-enterprise/3.2/guides/troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+And after setting that directive, [reload Sensu Enterprise](#reloading-configuration):
+
+{{< highlight shell >}}
+sudo systemctl reload sensu-enterprise{{< /highlight >}}
+
+Keep in mind that to set log levels back to normal, you can either run `sudo kill -TRAP $SENSUPID` (if you've used that method), or revert the change in `/etc/default/sensu-enterprise` and reload Sensu Enterprise for the change to take place.
+
+_NOTE: By default, Sensu's logging level is set to `info`. However, there are more log levels available than just `info` and `debug`. You can find the full list of available log levels in the [configuration reference documentation][4]. It's worth noting that `debug` is the most granular log level, while `fatal` is the least granular._
+
+### Printing Configurations
```
Comment (rgeniesse) [inline content/sensu-enterprise/3.2/guides/troubleshooting.md:77]: It is worth a note somewhere in here I think to not run that print config in `/etc/sensu/conf.d/`, as Sensu will pickup that as a configuration file as it ends in `.json` on the next reload or restart, causing some fun issues.

I'd suggest `cd`ing to `/tmp` first or just doing `tee /tmp/se-config.json` on the end instead.

The command also outputs it to `stdout`, so we may also want to call out that this file gets created, as some users may not even realize it was done at all and just copy `stdout`.

**Review by rgeniesse — APPROVED** [review]
💯 ! Looks great!

File: content/sensu-enterprise/3.2/guides/troubleshooting.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+### Resolving Private Key Errors
+
+If Sensu Enterprise attempts to load an SSL or TLS certificate in an unsupported format, you may see the following error:
+
+{{< highlight text >}}
+Unexpected exception: undefined method `get_private_key_info' for #
+An error was encountered while loading an SSL private key.
+{{< /highlight >}}
+
+You can resolve this issue by converting the certificate to plaintext using:
```
Comment (cwjohnston) [inline content/sensu-enterprise/3.2/guides/troubleshooting.md]: A couple of thoughts that might help to add some context here:

We expect to run into this error in the context of configuring SE for TLS/SSL connections to RabbitMQ or Redis, where we need a PEM-formatted certificate and private key. Typically we find that the error is caused by using an encrypted private key. 

As [shown here](https://github.com/kjur/jsrsasign/wiki/Tutorial-for-PKCS5-and-PKCS8-PEM-private-key-formats-differences), there are two possible formats for an encrypted PEM private key. Both formats are readily identifiable due to the presence of the word `ENCRYPTED`.

File: content/sensu-enterprise/3.2/guides/troubleshooting.md (hunk 4 above)
Comment (apaskulin) [reply content/sensu-enterprise/3.2/guides/troubleshooting.md:77]: Fixed in https://github.com/sensu/sensu-docs/pull/790/commits/6ed9b0d7f943a2064ddaa89891e48531b59977f8

File: content/sensu-enterprise/3.2/guides/troubleshooting.md (hunk 3 above)
Comment (apaskulin) [reply content/sensu-enterprise/3.2/guides/troubleshooting.md:49]: Fixed in https://github.com/sensu/sensu-docs/pull/790/commits/6ed9b0d7f943a2064ddaa89891e48531b59977f8

**Pull request merged**